### PR TITLE
Remove hostname test as it is not providing incremental value

### PR DESCRIPTION
**Pull request by nick-stroud**
We are already calling srun on each partition to get mount information. I don't believe we are getting enough incremental value out of also running hostname to justify the time it takes to run across each partition.

If #1004 is merged first then this PR will have to be rebased. 

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Review by cboneti — APPROVED** [review]
I would like you to add a comment to that test or rename the test (see my comment). However, you don't have to request another approval since it is a simple thing.

File: tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml:44]: Good point, this was my intention but calling it out is important for future maintainers. I will add this before submitting.

Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/test-partitions.yml]: Done

**Review by cboneti — APPROVED** [review]

**Pull request merged**
